graphql/end_to_end_test.go

```diff
@@ -201,3 +201,104 @@ func TestEndToEndAwaitAndCache(t *testing.T) {
 		t.Errorf("expected 4 total calls to slow, got %d", calls)
 	}
 }
+
+func TestArgumentOptionality(t *testing.T) {
+	schema := schemabuilder.NewSchema()
+	query := schema.Query()
+
+	query.FieldFunc("optional", func(args struct{ X *int64 }) int64 {
+		if args.X != nil {
+			return *args.X
+		}
+		return -1
+	})
+
+	_ = schema.Mutation()
+
+	builtSchema := schema.MustBuild()
+
+	// argument is optional and not passed in
+	q := graphql.MustParse(`
```

Comment:
can we clean up the copy/paste in an inner function to make this easier to read?